db.go

```diff
@@ -328,8 +472,19 @@ func Open(dir string, l log.Logger, r prometheus.Registerer, opts *Options) (db
 		minValidTime = blocks[len(blocks)-1].Meta().MaxTime
 	}
 
-	if err := db.head.Init(minValidTime); err != nil {
-		return nil, errors.Wrap(err, "read WAL")
+	if initErr := db.head.Init(minValidTime); initErr != nil {
+		// So that we can pick up errors even when wrapped.
+		if err := errors.Cause(initErr); err != nil {
```

Comment:
If `err` turns out to be `nil` (which I guess not), shouldn't we return `errors.Wrap(err, "read WAL")` like before?

Reply:
I refactored a bit to simplify this.